packages/yew/src/dom_bundle/btag/attributes.rs

```diff
@@ -163,9 +163,9 @@ impl Attributes {
 
     fn set(el: &Element, key: &str, value: &str, apply_as: ApplyAttributeAs) {
         match apply_as {
-            ApplyAttributeAs::Attribute => {
-                el.set_attribute(key, value).expect("invalid attribute key")
-            }
+            ApplyAttributeAs::Attribute => el
+                .set_attribute(intern(key), intern(value))
```

Comment:
I'm a bit worried that the interning of `value` will lead to confusion later down the road. Note that the benchmark has a relatively small value set here. In reality, values could also contain such thing as inline styles (for sub-pixel positions of draggable elements) and other often changing and unique values which don't benefit a lot from this caching. I'd rather expose this in some way in the `AttrValue` constructors to allow for customization and opt-in by the user.

Reply:
Makes perfect sense, this is why I kept these two commits separate and benched them separately. Basically interning attribute values helps if you have something like a table row with a class attribute that's being created many times, or for things like `type="text"`, and hurts, as you note, with things like animation via `style` attributes. I'm not familiar enough with Yew to know if it's feasible for users to opt into this in performance-sensitive situations.

Reply:
The value is stored as [`AttrValue`](https://docs.rs/yew/latest/yew/virtual_dom/type.AttrValue.html) which can have a `const INTERNED: bool` generic parameter. I'm not sure if that's enough though.

In any case, the user can just decide when passing the value as:
```rust
html! { <div style="styles"> <input type={intern("text")} /> </div> }
```
This way, the optimization can become opt-in.

Reply:
Does it even have to have a generic parameter? If, and that's how I'm reading the interning, all the user has to do is call `intern` on the passed value somewhere, it should just auto-magically get preferential treatment when crossing FFI boundaries (by a lookup in some map under the hood) without any additional action on our side. I think the code above works with yew being oblivious to the fact that `"text"` has been interned.

Reply:
Great point, if a user calls `intern(...)` in an attribute value it will intern. This makes it easy to opt in. (That is hard to do for attribute names and event types given the macro syntax, of course, but easy to do for attribute values.)